subsys/net/lib/lwm2m/lwm2m_obj_firmware_pull.c

```diff
@@ -51,6 +51,25 @@ firmware_udp_receive(struct net_app_ctx *app_ctx, struct net_pkt *pkt,
 	lwm2m_udp_receive(&firmware_ctx, pkt, true, NULL);
 }
 
+static void update_status_from_result(int result)
```

Comment:
I just realized this is terribly named.  We're setting the UPDATE_RESULT (not status) based on error code.  I'll post V2 with this renamed to: set_update_result_from_error()